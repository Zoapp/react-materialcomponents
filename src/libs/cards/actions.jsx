/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React, { Children } from "react";
import PropTypes from "prop-types";
import Rmdc from "../";

/**
 * mdc-card__actions
 * See
 * https://material.io/components/web/catalog/cards/
 *
 */

const MDC_CARD_ACTIONS = "mdc-card__actions";

const CardActions = ({
  children, vertical, ...props
}) => {
  let classes = MDC_CARD_ACTIONS;
  if (vertical) {
    classes += " mdc-card__actions--vertical";
  }
  // Iterate through children and set 'mdc-card__action'
  const ch = Children.map(children, child =>
    React.cloneElement(child, { ...child.props, cardAction: true }));
  return Rmdc.render(<section className={classes}>{ch}</section>, props);
};

CardActions.defaultProps = {
  mdcElement: MDC_CARD_ACTIONS,
  children: null,

  vertical: false,
  isactions: "true",
};

CardActions.propTypes = {
  mdcElement: PropTypes.string,
  children: PropTypes.node,

  vertical: PropTypes.bool,
  // Don't modify it
  isactions: PropTypes.string,
};

export default CardActions;
