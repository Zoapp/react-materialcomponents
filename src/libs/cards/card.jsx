/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React, { Children } from "react";
import PropTypes from "prop-types";
import CardPrimary from "./primary";
import Zrmc from "../";

/**
 * mdc-card
 * See
 * https://material.io/components/web/catalog/cards/
 * https://material-components-web.appspot.com/card.html
 *
 */

const MDC_CARD = "mdc-card";

const Card = ({
  children,
  title,
  subTitle,
  largeTitle,
  horizontalBlock,
  ...props
}) => {
  const classes = MDC_CARD;
  let primary = "";
  if (title || subTitle) {
    primary = (
      <CardPrimary title={title} subTitle={subTitle} largeTitle={largeTitle} />
    );
  }
  let ca = children;
  if (horizontalBlock) {
    // Extract other sections without CardActions from children
    const sections = Children.map(children, (child) => {
      if (child.props && child.props.isactions !== "true") {
        return child;
      }
      ca = React.cloneElement(child);
      return null;
    });
    primary = (
      <div className="mdc-card__horizontal-block">
        {primary}
        {sections}
      </div>
    );
  }
  return Zrmc.render(
    <div className={classes}>
      {primary}
      {ca}
    </div>,
    props,
  );
};

Card.defaultProps = {
  mdcElement: MDC_CARD,
  children: null,
  title: null,
  subTitle: null,
  largeTitle: false,
  horizontalBlock: false,
};

Card.propTypes = {
  mdcElement: PropTypes.string,
  children: PropTypes.node,
  title: PropTypes.string,
  subTitle: PropTypes.string,
  largeTitle: PropTypes.bool,
  horizontalBlock: PropTypes.bool,
};

export default Card;
