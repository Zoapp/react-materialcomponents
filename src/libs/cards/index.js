/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import CardActions from "./actions";
import Card from "./card";
import CardMedia from "./media";
import CardText from "./text";
import CardPrimary from "./primary";
import CardTitle from "./title";
import CardSubtitle from "./subtitle";

export {
  Card,
  CardActions,
  CardMedia,
  CardText,
  CardPrimary,
  CardTitle,
  CardSubtitle,
};
