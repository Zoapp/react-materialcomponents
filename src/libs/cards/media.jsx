/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React from "react";
import PropTypes from "prop-types";
import Zrmc from "../";

/**
 * mdc-card__media
 * See
 * https://material.io/components/web/catalog/cards/
 *
 */

const MDC_CARD_MEDIA = "mdc-card__media";

const CardMedia = ({
  children, src, size, alt, ...props
}) => {
  let classes = MDC_CARD_MEDIA;
  if (src) {
    classes += `-item mdc-card__media-item--${size}`;
  }

  // Check if src and set img instead
  let element;
  if (src) {
    element = (<img className={classes} src={src} alt={alt} style={props.style} />);
  } else {
    element = (<section className={classes}>{children}</section>);
  }
  return Zrmc.render(element, props);
};

CardMedia.defaultProps = {
  mdcElement: MDC_CARD_MEDIA,
  children: null,
  src: null,
  size: "1x",
  alt: "",
};

CardMedia.propTypes = {
  mdcElement: PropTypes.string,
  children: PropTypes.node,
  src: PropTypes.string,
  // TODO validate size type: 1x , 1dot5x, 2x, 3x
  size: PropTypes.string,
  alt: PropTypes.string,
};

export default CardMedia;
