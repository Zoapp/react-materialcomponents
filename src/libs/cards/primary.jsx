/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React, { Children } from "react";
import PropTypes from "prop-types";
import CardTitle from "./title";
import CardSubtitle from "./subtitle";
import Rmdc from "../";

/*
mdc-card__primary
See
https://material.io/components/web/catalog/cards/

*/
const CardPrimary = ({
  children, className, title, subTitle, largeTitle, isTitlingSecond, ...props
}) => {
  let classes = "mdc-card__primary";
  if (className) {
    classes += ` ${className}`;
  }
  let t = "";
  let s = "";
  if (title) {
    t = (<CardTitle>{title}</CardTitle>);
  }
  if (subTitle) {
    s = (<CardSubtitle>{subTitle}</CardSubtitle>);
  }
  let first = "";
  let ch = children;
  if (isTitlingSecond) {
    ch = Children.map((child, index) => {
      if (index === 0) {
        first = React.cloneElement(child);
        return null;
      }
      return child;
    });
  }
  return Rmdc.render(<section className={classes} {...props}>{first}{t}{s}{ch}</section>, props);
};

CardPrimary.defaultProps = {
  children: null,
  className: null,

  title: null,
  subTitle: null,
  largeTitle: false,
  isTitlingSecond: false,
};

CardPrimary.propTypes = {
// React component props
  children: PropTypes.node,
  className: PropTypes.string,

  title: PropTypes.string,
  subTitle: PropTypes.string,
  largeTitle: PropTypes.bool,
  isTitlingSecond: PropTypes.bool,
};

export default CardPrimary;
