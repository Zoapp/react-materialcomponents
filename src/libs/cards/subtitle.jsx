/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React from "react";
import PropTypes from "prop-types";
import Zrmc from "../";

/**
 * mdc-card__subtitle
 * See
 * https://material.io/components/web/catalog/cards/
 *
 */

const MDC_CARD_SUBTITLE = "mdc-card__subtitle";

const CardSubtitle = ({ children, className, ...props }) => {
  const classes = MDC_CARD_SUBTITLE;
  return Zrmc.render(<h2 className={classes}>{children}</h2>, props);
};

CardSubtitle.defaultProps = {
  mdcElement: MDC_CARD_SUBTITLE,
  children: null,
};

CardSubtitle.propTypes = {
  mdcElement: PropTypes.string,
  children: PropTypes.node,
};

export default CardSubtitle;
