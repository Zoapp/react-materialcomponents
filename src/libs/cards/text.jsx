/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React from "react";
import PropTypes from "prop-types";
import Rmdc from "../";

/**
 * mdc-card__supporting-text
 * See
 * https://material.io/components/web/catalog/cards/
 *
 */

const MDC_CARD_TEXT = "mdc-card__supporting-text";

const CardText = ({ children, ...props }) => {
  const classes = MDC_CARD_TEXT;
  return Rmdc.render(<section className={classes}>{children}</section>, props);
};

CardText.defaultProps = {
  mdcElement: MDC_CARD_TEXT,
  children: null,
};

CardText.propTypes = {
  mdcElement: PropTypes.string,
  children: PropTypes.node,
};

export default CardText;
