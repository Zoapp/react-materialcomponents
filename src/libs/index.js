/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import Rmdc from "./rmdc";
import Button from "./components/button";
import Icon from "./components/icon";
import Content from "./components/content";
import Checkbox from "./components/checkbox";
import Fab from "./components/fab";
import TextField from "./components/textfield";
import Radio from "./components/radio";
import FormField from "./components/formField";
import Select from "./components/select";
import { Menu, MenuAnchor, MenuItem, MenuDivider } from "./menu";
import Snackbar from "./components/snackbar";
import LinearProgress from "./components/linearProgress";
import Switch from "./components/switch";
import Slider from "./components/slider";
import IconToggle from "./components/iconToggle";
import { Toolbar, ToolbarRow, ToolbarSection, ToolbarTitle, ToolbarIcon } from "./toolbar";
import { Drawer, DrawerContent, DrawerHeader } from "./drawer";
import { List, ListItem, ListItemMeta, ListDivider, ListGroup, ListGroupSubheader } from "./list";
import { Grid, Inner, Cell } from "./layoutGrid";
import { Card, CardActions, CardMedia, CardText, CardPrimary, CardTitle, CardSubtitle } from "./cards";
import { Tabbar, Tab } from "./tab";
import { GridList, Tile } from "./gridList";
import { Dialog, DialogHeader, DialogBody, DialogFooter } from "./dialog";

export {
  Button, Content, Fab, TextField, Checkbox, Radio, FormField, Select, Snackbar,
  LinearProgress, Switch, Slider, IconToggle, Icon,
  Toolbar, ToolbarRow, ToolbarSection, ToolbarTitle, ToolbarIcon,
  Drawer, DrawerContent, DrawerHeader,
  List, ListItem, ListItemMeta, ListDivider, ListGroup, ListGroupSubheader,
  Grid, Inner, Cell,
  Card, CardActions, CardMedia, CardText, CardPrimary, CardTitle, CardSubtitle,
  Tabbar, Tab,
  GridList, Tile,
  Dialog, DialogHeader, DialogBody, DialogFooter,
  Menu, MenuAnchor, MenuItem, MenuDivider,
};

export default Rmdc;
