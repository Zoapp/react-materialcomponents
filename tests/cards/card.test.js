import React from "react";
import renderer from "react-test-renderer";
import { Card } from "libs/cards/";

describe("components/cards/Card", () => {
  it("can have children", () => {
    const tree = renderer.create(<Card>Children are here!</Card>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can have a title", () => {
    const tree = renderer.create(<Card title="foo" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can have a subtitle", () => {
    const tree = renderer.create(<Card subTitle="bar" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can have an horizontal block", () => {
    const tree = renderer.create(<Card horizontalBlock />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
