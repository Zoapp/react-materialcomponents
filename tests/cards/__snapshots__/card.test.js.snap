// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/cards/Card can have a subtitle 1`] = `
<div
  className="mdc-card"
>
  <section
    className="mdc-card__primary"
  >
    
    
    <h2
      className="mdc-card__subtitle"
    >
      bar
    </h2>
  </section>
</div>
`;

exports[`components/cards/Card can have a title 1`] = `
<div
  className="mdc-card"
>
  <section
    className="mdc-card__primary"
  >
    
    <h1
      className="mdc-card__title"
    >
      foo
    </h1>
    
  </section>
</div>
`;

exports[`components/cards/Card can have an horizontal block 1`] = `
<div
  className="mdc-card"
>
  <div
    className="mdc-card__horizontal-block"
  >
    
  </div>
</div>
`;

exports[`components/cards/Card can have children 1`] = `
<div
  className="mdc-card"
>
  
  Children are here!
</div>
`;
